src/telliot_feeds/sources/price/historical/kraken_ohlc.py

```diff
@@ -0,0 +1,66 @@
+from dataclasses import dataclass
+from typing import Any
+from typing import Optional
+from urllib.parse import urlencode
+
+import pandas as pd
+
+from telliot_feeds.pricing.price_source import PriceSource
+from telliot_feeds.utils.log import get_logger
+from telliot_feeds.sources.price.historical.kraken import KrakenHistoricalPriceService
+
+logger = get_logger(__name__)
+
+
+# Hardcoded supported assets & currencies
+# Kraken uses XBT instead of BTC for its APIs:
+# https://support.kraken.com/hc/en-us/articles/360001206766-Bitcoin-currency-code-XBT-vs-BTC
+kraken_assets = {"ETH", "XBT"}
+kraken_currencies = {"USD"}
+
+
+class KrakenHistoricalPriceServiceOHLC(KrakenHistoricalPriceService):
+
+    def get_request_url(self, asset: str, currency: str, period_start: int) -> str:
+        """Assemble Kraken historical trades request url."""
+        asset = asset.upper()
+        currency = currency.upper()
+
+        if asset not in kraken_assets:
+            raise Exception(f"Asset not supported: {asset}")
+        if currency not in kraken_currencies:
+            raise Exception(f"Currency not supported: {currency}")
+
+        url_params = urlencode({"pair": f"{asset}{currency}", "since": period_start, "interval": 1440})
+        # Source: https://docs.kraken.com/rest/#operation/getRecentTrades
+        return f"/0/public/OHLC?{url_params}"
+
+    def resp_price_parse(self, asset: str, currency: str, resp: dict[Any, Any]) -> Optional[float]:
+        """Gets first price from trades data."""
+        try:
+            # Price of last trade in trades list retrieved from API
+            pair_key = f"X{asset.upper()}Z{currency.upper()}"
+            data = resp["result"][pair_key]
+            df = pd.DataFrame(data, columns=["CloseTime", "Open", "High", "Low", "Close", "?", "V", "QV"])
+            df["Close"] = df["Close"].astype(float)
+            volatility = df["Close"].pct_change().std()
+        except KeyError as e:
```

Comment:
catch `KeyError` for only the line/lines you expect to throw that error, instead of putting a bunch of lines in a try/catch block. This makes it easier for others to debug in the future. Also, after catching your expected errors like `KeyError`, catch all other errors with `except Exception as e` and log 'em